php/class-post-type.php

```diff
@@ -549,4 +552,86 @@ public function filter_user_has_cap( $allcaps, $caps ) {
+		if ( did_action( 'customize_save' ) ) {
+			// Short circuit because customize_save ajax call is changing status.
+			return;
+		}
+		$snapshot_theme = get_post_meta( get_the_ID(), '_snapshot_theme', true );
+		if ( ! empty( $snapshot_theme ) && get_stylesheet() !== $snapshot_theme ) {
+			// Theme mismatch.
+			if ( defined( 'DOING_CRON' ) && DOING_CRON ) {
+				wp_update_post( array(
+					'ID' => $post->ID,
+					'post_status' => 'pending',
+				) );
+			}
+		}
+		$this->snapshot_manager->ensure_customize_manager();
+
+		if ( ! did_action( 'customize_register' ) ) {
+			/*
+			 * When running from CLI or Cron, we have to remove the action because
+			 * it will get added with a default priority of 10, after themes and plugins
+			 * have already done add_action( 'customize_register' ), resulting in them
+			 * being called first at the priority 10. So we manually call the
+			 * prerequisite function WP_Customize_Manager::register_controls() and
+			 * remove it from being called when the customize_register action fires.
+			 */
+			remove_action( 'customize_register', array( $this->snapshot_manager->customize_manager, 'register_controls' ) );
+			$this->snapshot_manager->customize_manager->register_controls();
+
+			/** This action is documented in wp-includes/class-wp-customize-manager.php */
+			do_action( 'customize_register', $this->snapshot_manager->customize_manager );
+		}
+		$snapshot_content = $this->get_post_content( $post );
+		$snapshot_values = array_filter(
+			wp_list_pluck( $snapshot_content, 'value' ),
+			function( $value ) {
+				return ! is_null( $value );
+			}
+		);
+
+		if ( method_exists( $this->snapshot_manager->customize_manager, 'validate_setting_values' ) ) {
+			/** This action is documented in wp-includes/class-wp-customize-manager.php */
+			do_action( 'customize_save_validation_before', $this->snapshot_manager->customize_manager );
+		}
+
+		$this->snapshot_manager->customize_manager->add_dynamic_settings( array_keys( $snapshot_values ) );
+		$setting_objs = array();
+		foreach ( $snapshot_values as $setting_id => $setting_params ) {
+			$this->snapshot_manager->customize_manager->set_post_value( $setting_id, $setting_params );
+			$setting_objs[] = $this->snapshot_manager->customize_manager->get_setting( $setting_id );
+		}
+
+		$filtered_setting_objs = array_filter( $setting_objs, function( $setting_obj ) {
+			return is_a( $setting_obj, '\WP_Customize_Setting' );
+		} );
+		if ( empty( $filtered_setting_objs ) ) {
+			return;
+		}
+
+		foreach ( $filtered_setting_objs as $setting_obj ) {
+			$setting_obj->capability = 'exist';
+		}
+
+		remove_action( 'customize_save', array( $this->snapshot_manager, 'check_customize_publish_authorization' ), 10 );
+		/** This action is documented in wp-includes/class-wp-customize-manager.php */
+		do_action( 'customize_save', $this->snapshot_manager->customize_manager );
+		add_action( 'customize_save', array( $this, 'check_customize_publish_authorization' ), 10, 0 );
```

Comment:
Looks like the wrong `customize_save` handler is being re-added. There is no `check_customize_publish_authorization` method on this class.

I suggest doing:

```php
$action = 'customize_save';
$callback = array( $this->snapshot_manager, 'check_customize_publish_authorization' );
$priority = has_action( $action, $callback );
if ( false !== $priority ) {
    remove_action( $action, $callback, $priority );
}
/** This action is documented in wp-includes/class-wp-customize-manager.php */
do_action( 'customize_save', $this->snapshot_manager->customize_manager );
if ( false !== $priority ) {
    add_action( $action, $callback, $priority );
}
```